### PR TITLE
dhcpclient Auth Fix

### PR DESCRIPTION
**Pull request by marjohn56**
This should do it. 

Now, need to move onto dhcp6c and the RAW stuff.

### CODE REVIEW

**Comment by fichtner** [comment]
What is it supposed to do? Do we need something in the config?

**Comment by marjohn56** [comment]
rfc3118-auth is a requirement in the send options

**Comment by marjohn56** [comment]
I'll email you what is required  to be sent, I cannot post it here as it's Kev's login :)

**Comment by marjohn56** [comment]
I've sent a message to lastsummer.

**Comment by fichtner** [comment]
Okay, maybe a silly question: does specifying "option-90" not work so that it has to be "rfc3118-auth" with this patch?

File: sbin/dhclient/conflex.c [hunk 1]
```diff
@@ -467,6 +467,8 @@ intern(char *atom, int dfv)
 			return (REBOOT);
 		if (!strcasecmp(atom + 1, "eject"))
 			return (REJECT);
+		if (!strcasecmp(atom + 1, "fc3118-auth"))
```
Comment (fichtner) [inline sbin/dhclient/conflex.c]: This probably has no effect, it returns the wrong token anyway (REJECT)

Comment (marjohn56) [reply sbin/dhclient/conflex.c]: Strange, as it works perfectly and it doesn't without it. 

I have no idea if option 90 in place of fc3118-auth works, I'll try it later.

Comment (fichtner) [reply sbin/dhclient/conflex.c]: dhclient is weird oO

Comment (marjohn56) [reply sbin/dhclient/conflex.c]: option-90 does work. So if that be the case, then job is done. :)

All I did to test that was replace the "rfc3118-auth", "X"... in tables.c and recompile.

Makes one ask, why the hell put it in the original diff?

Comment (fichtner) [reply sbin/dhclient/conflex.c]: Not enough time to understand dhclient I think. Like the BPF filter, too hard, XXX do later. But I can somewhat relate :)

**Comment by fichtner** [comment]
@marjohn56 so what bits do we need here? hopefully none? :)

**Comment by marjohn56** [comment]
Yeah, option-90 seems to work fine so just leave as is. I tested it and it appears to work, I'll do a clean compile without the changes and send it to Kev for testing. What to we need to do to get dhcp6c RAW options ready, or does that already have all your updates in?

**Comment by fichtner** [comment]
dhcp6c raw options pending review, merge and release in 18.1.x. I'll get the review and merge done this week.

**Comment by marjohn56** [comment]
That will make Kev sooo happy. :)

Will also mean an influx of Orange France users I suspect.

**Comment by fichtner** [comment]
yeah, we should have a doc for them like what you wrote for yours: https://docs.opnsense.org/manual/how-tos/IPv6_ZenUK.html

otherwise everybody could use it but nobody will know

**Comment by marjohn56** [comment]
Kev, can you do a doc for this and I'll the do a how to for the manual.

**Comment by marjohn56** [comment]
I think we need two documents. One specifically for Orange France which says how to set up both v6 and v4 and another to show how dhcp6c raw options work. dhclient per say does not really need a separate document as that will be part of the man for it.

**Comment by fichtner** [comment]
Sounds like a plan, thanks :)

**Comment by nivek1612** [comment]
This is getting exciting - I already have a doc I used to explain how to use pfSense (sorry) in France so happy to do the same for OPNsense once we have it all finalised. I may be making a trip out sooner than I thought as I lost all internet in France this morning. Although I can't even sign into the Orange client web so I think its not a local issue :-)

**Comment by nivek1612** [comment]
User error caused the loss of option 55 - whoops
 
testing with "option-90" rather than "rfc3118-auth" works in the latest compile. I have a dhcp request that matches perfectly with ORANGE FTTP requirements  

Nice work chaps

**Comment by marjohn56** [comment]
Cool, that's a wrap them.

**Comment by marjohn56** [comment]
It's not a wrap... Whilst messing with dhcp6c I saw a log error from dhclient, 

`dhclient[79819]: /var/etc/dhclient_wan.conf line 9: rfc3118-auth: unexpected option name.`

It's also a request option, so now we know why it's there... I will recompile with this patch and see if it does in fact fix it.

**Comment by nivek1612** [comment]
just change the request in the config to option-90 
Thats what caused option-55 to disappear for me earlier because i left rfc3118-auth in the request options field 

no need to recompile unless I'm being stupid

**Comment by marjohn56** [comment]
OK, tested with this patch in place, now i see this.

image

so it does need this patch for the param request list.

**Comment by fichtner** [comment]
And without? Not sure what we're looking at.

**Comment by marjohn56** [comment]
kev check it on yours and ensure the para request list is correct

**Comment by nivek1612** [comment]
looks the same to me 

screen shot 2018-06-13 at 18 12 29

**Comment by nivek1612** [comment]
whats are you setting in here in the GUI
screen shot 2018-06-13 at 09 05 50

**Comment by fichtner** [comment]
The only difference I see is the send priority of 77 and 90, that's what the patch moves up but it shouldn't be strictly needed

https://github.com/opnsense/src/pull/30/files#diff-3884f632851d57585461904822bc58faR338

**Comment by marjohn56** [comment]
auth blah blah .. try it with 'option-90'

**Comment by marjohn56** [comment]
Its the parameter request list item  (90)

**Comment by marjohn56** [comment]
Having issues with that... I've taken the bits from the RAW branch and added them to the main branch, it will nor play, refuses to send the options. I'll go through it again after dinner, but it's the last chance i'll likely get this week as real work beckons.

**Comment by marjohn56** [comment]
OK, dhcp6c sorted I think, Kev has a test binary, working on my system, My issue was a couple of lines that were not in dhcp6c.c, had to look at the original diff file to find the little b******.

So it's @fichtner dhcp6c lightweight client with the RAW option support.

**Comment by nivek1612** [comment]
just test and looks perfect

Should be in France on Saturday and able to test both dhclient and dhcp6c IRL

**Pull request closed without merging**